internal/controller/controllers/agent_controller.go

```diff
@@ -113,6 +121,20 @@ func (r *AgentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 	return r.updateStatus(ctx, agent, host, nil, false)
 }
 
+func (r *AgentReconciler) deleteAgent(ctx context.Context, agent types.NamespacedName) (ctrl.Result, error) {
+	agentToDelete := &aiv1beta1.Agent{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      agent.Name,
+			Namespace: agent.Namespace,
+		},
+	}
+	if delErr := r.Client.Delete(ctx, agentToDelete); delErr != nil {
+		r.Log.WithError(delErr).Errorf("Failed to get delete resource %s %s", agent.Name, agent.Namespace)
+		return ctrl.Result{Requeue: true}, delErr
```

Comment:
controller runtime does track metrics for error rates, so in theory if you'd like this problem to show up in those (as you might have alerts on them), it might be good to return an error if you're requeuing because of an error.

Reply:
Good point. I tend to think that metric isn't particularly useful, but it's subjective. Here's a longer discussion on the topic: https://github.com/kubernetes-sigs/controller-runtime/issues/617

Reply:
Changed to return `nil` and Requeue set `true` only if we had an error deleting that agent.